### PR TITLE
NOISSUE - Add name field for Bootstrap Config

### PR DESCRIPTION
**Pull request by dborovcanin**
### What does this do?
This PR adds a `name` field to the Bootstrap service Config data model. This field represents the human-readable name of the Config and is not required. Bootstrap service supports case-insensitive, partial-match searching over this field.

### List any changes that modify/break current functionality
There are no such changes.

### Have you included tests for your changes?
Yes.

### Did you document any new/modified functionality?
Yes, API docs are updated.

### CODE REVIEW

File: bootstrap/api/endpoint.go [hunk 1]
```diff
@@ -128,34 +131,43 @@ func listEndpoint(svc bootstrap.Service) endpoint.Endpoint {
...
-				channels = append(channels, channelRes{
-					ID:       ch.ID,
-					Name:     ch.Name,
-					Metadata: ch.Metadata,
-				})
+		switch {
+		case req.filter.Unknown:
+			res := listUnknownRes{}
+			for _, cfg := range configs {
+				res.Configs = append(res.Configs, unknownRes{ExternalID: cfg.ExternalID, ExternalKey: cfg.ExternalKey})
```
Comment (manuio) [inline bootstrap/api/endpoint.go]: fix the tabulation

File: bootstrap/api/endpoint_test.go [hunk 2]
```diff
@@ -527,6 +534,13 @@ func TestList(t *testing.T) {
 			status: http.StatusOK,
 			res:    list[0:1],
 		},
+		{
+			desc:   "view list searching by name",
+			auth:   validToken,
+			url:    fmt.Sprintf("%s/configs?offset=%d&limit=%d&name=%s", bs.URL, 0, 100, "95"),
```
Comment (manuio) [inline bootstrap/api/endpoint_test.go:540]: can you create const for this values?

File: bootstrap/mocks/configs.go [hunk 3]
```diff
@@ -73,15 +74,20 @@ func (crm *configRepositoryMock) RetrieveAll(key string, filter bootstrap.Filter
...
 
+	if s, ok := filter.PartialMatch["name"]; ok {
+		name = strings.ToLower(s)
+	}
+
 	for _, v := range crm.configs {
 		id, _ := strconv.ParseUint(v.MFThing, 10, 64)
 		if id >= first && id < last {
-			if (state == -1 || v.State == state) && v.Owner == key {
+			if (state == -1 || v.State == state) && (name == "" || strings.Index(strings.ToLower(v.Name), name) != -1) && v.Owner == key {
```
Comment (manuio) [inline bootstrap/mocks/configs.go]: Maybe you can split this checks

File: bootstrap/service_test.go [hunk 4]
```diff
@@ -283,22 +284,22 @@ func TestList(t *testing.T) {
 			err:    nil,
 		},
 		{
-			desc:   "list config unauthorized",
-			config: []bootstrap.Config{},
-			filter: bootstrap.Filter{},
-			key:    invalidToken,
+			desc:   "list configs with specified name",
+			config: saved[95:96],
```
Comment (manuio) [inline bootstrap/service_test.go:288]: Maybe use const here also

File: bootstrap/service_test.go [hunk 5]
```diff
@@ -283,22 +284,22 @@ func TestList(t *testing.T) {
...
+			filter: bootstrap.Filter{PartialMatch: map[string]string{"name": "95"}},
+			key:    validToken,
 			offset: 0,
-			limit:  10,
-			err:    bootstrap.ErrUnauthorizedAccess,
+			limit:  100,
```
Comment (manuio) [inline bootstrap/service_test.go:292]: and here

**Review by manuio — CHANGES_REQUESTED** [review]

**Review by drasko — CHANGES_REQUESTED** [review]

File: bootstrap/service_test.go (hunk 4 above)
Comment (manuio) [reply bootstrap/service_test.go:288]: @drasko that's why the "maybe"

File: bootstrap/api/endpoint.go (hunk 1 above)
Comment (dborovcanin) [reply bootstrap/api/endpoint.go]: Tabulation is fine here.

**Comment by dborovcanin** [comment]
@manuio All the literals that are not extracted to `const` are test specific or test-case specific, so it's OK to keep it this way.

File: bootstrap/api/endpoint.go (hunk 1 above)
Comment (manuio) [reply bootstrap/api/endpoint.go]: No, in the same file you did it right:
```
channels = append(channels, channelRes{
	ID:       ch.ID,
	Name:     ch.Name,
	Metadata: ch.Metadata,
})
```

File: bootstrap/api/transport.go [hunk 6]
```diff
@@ -285,10 +287,13 @@ func parsePagePrams(q url.Values) (uint64, uint64, error) {
 }
 
 func parseFilter(values url.Values) bootstrap.Filter {
-	ret := bootstrap.Filter{}
+	ret := bootstrap.Filter{FullMatch: make(map[string]string), PartialMatch: make(map[string]string)}
```
Comment (manuio) [inline bootstrap/api/transport.go]: same here

File: bootstrap/postgres/configs.go [hunk 7]
```diff
@@ -255,26 +263,32 @@ func (cr configRepository) RemoveUnknown(key, id string) error {
...
 	return cr.db.Query(fmt.Sprintf(template, f), params...)
 }
 
-func toDBChannels(channels []bootstrap.Channel) []writeCh {
-	ret := []writeCh{}
+func toDBChannels(channels []bootstrap.Channel) []dbChannel {
+	ret := []dbChannel{}
 	for _, ch := range channels {
-		c := writeCh{ID: ch.ID, Name: ch.Name, Metadata: ch.Metadata}
+		c := dbChannel{ID: ch.ID, Name: ch.Name, Metadata: ch.Metadata}
```
Comment (manuio) [inline bootstrap/postgres/configs.go]: and here

**Review by manuio — CHANGES_REQUESTED** [review]

**Review by manuio — DISMISSED** [review]
LGTM

**Review by drasko — APPROVED** [review]
LGTM

**Review by anovakovic01 — APPROVED** [review]
LGTM!

**Pull request merged**